add windows window border color effect

amasondev opened the pull request:
<img width="855" alt="Screenshot 2023-08-10 225615" src="https://github.com/focus-editor/focus/assets/5068750/ab3b05f9-b643-4e1f-83d7-0643c13050c3">

This makes the editor window look a lot cleaner on Windows. Requires bringing in DwmSetWindowAttribute from Dwmapi, which is not yet defined in jai's Windows module. Stubbed in noop procedures for the other platforms. I can take a look at a macOS implementation too. MacVim has it.

focus-editor commented:
OMG it looks awesome.

If only you'd sent it 30 minutes earlier, it would be a perfect addition to the 0.2.0 release.

I'll test it

focus-editor commented:
![image](https://github.com/focus-editor/focus/assets/119373822/4aed3fcd-1f67-405b-a229-a179586c71d1)

Doesn't seem to work for me. What could be the reason?

amasondev commented:
> ![image](https://user-images.githubusercontent.com/119373822/259921543-4aed3fcd-1f67-405b-a229-a179586c71d1.png)
> 
> Doesn't seem to work for me. What could be the reason?

https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/ne-dwmapi-dwmwindowattribute

> DWMWA_CAPTION_COLOR
> Use with [DwmSetWindowAttribute](https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/nf-dwmapi-dwmsetwindowattribute). Specifies the color of the caption. The pvAttribute parameter points to a value of type [COLORREF](https://learn.microsoft.com/en-us/windows/win32/gdi/colorref).
> 
> Specifying DWMWA_COLOR_DEFAULT (value 0xFFFFFFFF) for the color will reset the window back to using the system's default behavior for the caption color.
> 
> This value is supported starting with Windows 11 Build 22000.

Seems to only be supported on Windows >11

amasondev commented:
It's possible to at least use a dark title bar on 10 using DWMWA_USE_IMMERSIVE_DARK_MODE, which needs to have a value of 19 on some old versions and 20 on newer versions 🤷

focus-editor commented:
Tested on Windows 11, looks amazing.

If you add that dark title bar option we could include that as well

amasondev commented:
Added the dark mode setting, and added config options for both of these. I also removed the setting of the border color, as it tended to lose the contrast against other windows.

focus-editor commented:
Thanks. I will merge and rename the options to make them a bit more descriptive if you don't mind.

Also, the dark titlebar looks really terrible and has a bug when you just start the editor (resizing the window fixes it)
![image](https://github.com/focus-editor/focus/assets/119373822/e1b72dec-2ab3-42a8-afbd-7d38fb26475f)

I think it's best if we don't make it default

amasondev commented:
Oof, yeah sorry I wish I had a W10 VM handy for testing.
